compose/snippets/src/main/java/com/example/compose/snippets/topcomponents/AppBar.kt

```diff
@@ -0,0 +1,327 @@
+            else -> AppBarOptions(
+                toBottom = { selection = "bottomBar" },
+                toTopBarSmall = { selection = "topBar" },
+                toTopBarCenter = { selection = "topBarCenter" },
+                toTopBarMedium = { selection = "topBarMedium" },
+                toTopBarLarge = { selection = "topBarLarge" },
+            )
+        }
+    }
+}
+
+@Composable
+fun AppBarOptions(
+    toBottom: () -> Unit,
+    toTopBarSmall: () -> Unit,
+    toTopBarCenter: () -> Unit,
+    toTopBarMedium: () -> Unit,
+    toTopBarLarge: () -> Unit,
+) {
+    Column() {
+        Button({ toBottom() }) {
+            Text("Bottom bar")
+        }
+        Button({ toTopBarSmall() }) {
+            Text("Small top bar")
+        }
+        Button({ toTopBarCenter() }) {
+            Text("Center aligned top bar")
+        }
+        Button({ toTopBarMedium() }) {
+            Text("Medium top bar")
+        }
+        Button({ toTopBarLarge() }) {
+            Text("Large top bar")
+        }
+    }
+}
+
+@Composable
+fun BottomAppBarExample() {
+    Scaffold(
+        bottomBar = {
+            BottomAppBar(
+                actions = {
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(Icons.Filled.Check, contentDescription = "Localized description")
+                    }
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(
+                            Icons.Filled.Edit,
+                            contentDescription = "Localized description",
+                        )
+                    }
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(
+                            Icons.Filled.Mic,
+                            contentDescription = "Localized description",
+                        )
+                    }
+                    IconButton(onClick = { /* doSomething() */ }) {
+                        Icon(
+                            Icons.Filled.Image,
+                            contentDescription = "Localized description",
+                        )
+                    }
+                },
+                floatingActionButton = {
+                    FloatingActionButton(
+                        onClick = { /* do something */ },
+                        containerColor = BottomAppBarDefaults.bottomAppBarFabColor,
+                        elevation = FloatingActionButtonDefaults.bottomAppBarFabElevation()
+                    ) {
+                        Icon(Icons.Filled.Add, "Localized description")
+                    }
+                }
+            )
+        },
+    ) { innerPadding ->
+        Text(
+            modifier = Modifier.padding(innerPadding),
+            text = "Example of a scaffold with a bottom app bar."
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SmallTopAppBarExample() {
```

Comment:
For each of these that don't take parameters, please annotate with `@Preview` too so it shows up in design mode too.

Reply:
Done.